vms/platformvm/warp/README.md

```diff
@@ -0,0 +1,121 @@
+- `signature` is an aggregated BLS Multi-Signature of the Unsigned Message
+
+BitSetSignatures are verified within the context of a specific P-Chain height. At a specific P-Chain height, the PlatformVM serves a canonically ordered validator set for the source subnet (validator set is ordered lexicographically by the BLS public key bytes). The signers bitset encodes which validators' signature was included. A value of 1 at index i in `signers` claims that a signature from the validator at index i in the canonical validator set is included in the aggregate signature.
+
+The bitset tells the verifier which BLS public keys should be aggregated to verify the warp message.
+
+Signed Message:
+```
++------------------+------------------+-------------------------------------------------+
+| unsigned_message :  UnsignedMessage |                          size(unsigned_message) |
++------------------+------------------+-------------------------------------------------+
+|        signature :        Signature |                                 size(signature) |
++------------------+------------------+-------------------------------------------------+
+                                      |  size(unsigned_message) + size(signature) bytes |
+                                      +-------------------------------------------------+
+```
+
+## Sending an Avalanche Warp Message
+
+A Blockchain on Avalanche sends an Avalanche Warp Message by coming to agreement on the message that every validator should be willing to sign. As an example, the VM of a Blockchain may define that once a block is accepted, the VM should be willing to sign a message including the block hash in the payload to attest to any other Subnet that the block was accepted. The contents of the payload, how to aggregate the signature (VM-to-VM communication, off-chain relayer, etc.), is left to the VM.
+
+Once the validator set of a blockchain is willing to sign the message M, an aggregator performs the following process:
+
+1. Gather signatures of the message M from N validators (where the N validators meet the required threshold of stake on the destination chain)
+2. Aggregate the N signatures into a multi-signature
+3. Look up the canonical validator set at the P-Chain height where the message will be verified
+4. Encode the selection of the N validators included in the signature in a bitset
+5. Construct the Signed Message from the aggregate signature, bitset, and unsigned message
+
+## Verifying / Receiving an Avalanche Warp Message
+
+Avalanache Warp Messages are verified within the context of a specific P-Chain height included in the [ProposerVM](../../proposervm/README.md)'s header. The P-Chain height is provided as context to the wrapped VM when verifying the VM's internal blocks (implemented by the optional interface [WithVerifyContext](../../../snow/engine/snowman/block/block_context_vm.go)).
```

Comment:
Wrapped is probably correct when talking about it from a code perspective but maybe underlying is more clear to people who aren't as technical
```suggestion
Avalanache Warp Messages are verified within the context of a specific P-Chain height included in the [ProposerVM](../../proposervm/README.md)'s header. The P-Chain height is provided as context to the underlying VM when verifying the underlying VM's blocks (implemented by the optional interface [WithVerifyContext](../../../snow/engine/snowman/block/block_context_vm.go)).
```